sparse/coo/umath.py

```diff
@@ -173,7 +169,7 @@ def _elemwise_n_ary(func, *args, **kwargs):
     result_shape = _get_nary_broadcast_shape(*[arg.shape for arg in args])
 
     # Concatenate matches and mismatches
-    data = np.concatenate(data_list) if len(data_list) else np.empty((0,), dtype=args[0].dtype)
+    data = np.concatenate(data_list) if len(data_list) else np.empty((0,), dtype=func_value.dtype)
```

Comment:
Useful for comparisons, they can have any input `dtype`. But the output will always be the function value.